src/Components/DrillListPage.js

```diff
@@ -13,27 +14,39 @@ const DrillListPage = props => {
   const type = props.route.params.type;
   const drills = props.drills.filter(drill => drill.type === type);
 
+  const [data, setData] = useState(drills);
+  const [displayFilters, setDisplayFilters] = useState(false);
+
   return (
     <View style={styles.drillListPage}>
-      <Text style={styles.counter}>{drills.length} drills available</Text>
-      <FlatList
-        data={drills}
-        keyExtractor={item => item.id.toString()}
-        renderItem={({ item }) => (
-          <TouchableOpacity style={styles.drill} onPress={() => navigation.navigate('DrillPage', { drill: item })}>
-            <Image style={styles.image} source={{ uri: item.image }} />
-            <View style={styles.contentContainer}>
-              <Text style={styles.source}>{item.source}</Text>
-              <Text style={styles.title}>{item.title}</Text>
-              <Text style={styles.numberOfPlayers}>
-                Duration: {item.durationInMinutes} min - players: {item.nbPlayers}
-              </Text>
-            </View>
-          </TouchableOpacity>
-        )}
-        onEndReachedThreshold={0.5}
-        onEndReached={() => {}}
-      />
+      <Text style={styles.counter}>{data.length} drills available</Text>
+      {displayFilters ? (
+        <Filters
+          onConfirm={() => setDisplayFilters(false)}
+          onFiltered={drills => setData(drills)}
```

Comment:
My idea here is that the responsibility for filtering should not be on the List component but on the Filters (they know what are the filters)

Also this mean we can extend the code adding filters with no impact on this component?

But I'm not sure we can test the behaviour of one without the other one?

Any thoughts on this?